tests/src/main/java/com/squareup/duktape/tests/OctaneActivity.java

```diff
@@ -0,0 +1,68 @@
+package com.squareup.duktape.tests;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.widget.TextView;
+import com.squareup.duktape.Duktape;
+import java.io.IOException;
+import okio.BufferedSource;
+import okio.Okio;
+
+public final class OctaneActivity extends Activity implements Runnable {
+  private TextView output;
+  private View run;
+
+  @Override protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.octane);
+
+    output = (TextView) findViewById(R.id.output);
+
+    run = findViewById(R.id.run);
+    run.setOnClickListener(new OnClickListener() {
+      @Override public void onClick(View v) {
+        run.setEnabled(false);
+        output.setText("Running...\n\nThis will take a long time...");
+
+        // Schedule benchmark on next frame so updates can happen.
+        run.post(OctaneActivity.this);
+      }
+    });
+  }
+
+  @Override public void run() {
+    String benchmark;
+    try {
+      benchmark = benchmark();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    run.setEnabled(true);
+    output.setText(benchmark);
+  }
+
+  private String benchmark() throws IOException {
+    Duktape duktape = Duktape.create();
+
+    for (String file : getAssets().list("octane")) {
```

Comment:
might also be worthwhile measuring the time to evaluate these assets. We care both about parse time & execution time.